### ISSUE TITLE
release bulk_extractor 2.0.0 beta 1 on windows

### ISSUE THREAD

**Issue by simsong** [simsong/bulk_extractor#268]

**Comment by bpieczko** [comment]
A GUI would be great, but they can deal with a command line interface if
needed. Thanks.

**Comment by simsong** [comment]
Yes, if you have a student who would like to do this as an independent study project, I'm happy to assist in supervising. In the meantime, I'll create an executable and test it out. I also need to update the roadmap.

**Comment by simsong** [comment]
I have uploaded the command-line version of the windows executable. You can find it [here](https://github.com/simsong/bulk_extractor/releases/edit/v2.0.0-beta1).

**Comment by jsouthTo** [comment]
Hi - i can also manage the command line, but would like to use the java interface, I find it to be useful for reviewing the data after carving. Just easier for me to digest in that format for whatever reason. Saw the post on the google group saying to post on this issue if we wanted the JAR thanks! fantastic software

**Comment by simsong** [comment]
@jsouthTo  - Thanks for the comment. The 2.0 version should work with the existing Java user interface. However, I have not created a new installer yet. What platform are you using?

**Comment by jsouthTo** [comment]
Ah, okay, thank you. I've not been able to compile the JAR file on Kali (I
think I read in the docs just yesterday that Kali wasn't supported?) nor on
my Mac. I have a windows machine I use as well and I have been able to use
it on that machine, I just rarely use that one except for testing. I always
get an error regarding a JDK deficiency.

Install references an etc folder that isn't there in the non-release
builds, when I execute ./configure && make && make install there are
compilation errors. The config.sub that I thought may help indicate the
system type, doesn't appear to exist in this build.

I'm sure it will be something small and "easy" that I'm missing.

Thanks for the reply!

-Jimmy

**Comment by jsouthTo** [comment]
Sorry, didn't clarify my thought - distracted mid-email. I always got the
JDK deficiency message compiling on Mac or Kali.

That is not the error I saw this time.

**Comment by simsong** [comment]
> Sorry, didn't clarify my thought - distracted mid-email. I always got the JDK deficiency message compiling on Mac or Kali. That is not the error I saw this time.

Can you post the error message that you are seeing? Which version of JDK are you using? 
This is one of the reasons that we separated the GUI from the command-line version, BTW.
You shouldn't need to recompile the JAR; the old JAR should still work.

**Comment by jsouthTo** [comment]
Had some other things get in the way.

So bulk_extractor did compile on my Mac. The homebrew install I had, which I forgot about, was interfering. I was running BE just fine, and about 10 mins in I’m getting a disk write error. Never seen that happen before. The message is boiler plate and says it’s likely it’s a disk space issue, but I have 500GB free. I’ve zapped the data and tried again, twice.

The JAR, forgive me I’m not as update on Java as I used to be, can I take the windows JAR and run it on Mac or Linux? Per below, I’ve never been able to compile the JAR on any version of BE. I only have the binaries built ready to go for Windows. I couldn’t recall if a jar could be used on different systems and architectures (Mac is ARM)

Any thoughts? I couldn’t find anything with a quick google search.

Thanks!

-Jimmy Sobeck
416-529-4064

**Comment by simsong** [comment]
Please let me know more details of the disk write error. Can you run `df`, then run `bulk_extractor`, then run `df` again, and post the results here (or email them to me), and also post the `report.xml` file (or email it to me). The xml file has in it the name of the disk image but it doesn't have any information extracted from the disk image. It's easy to validate by eye if you want to look at it.

Java `jar` files are both OS and CPU independent — that is the point of java. At least, they are always independent unless there is use of JNI (Java Native Interface). JAR files compiled in the past still run today, but old java code may be hard to get to compile.

The bulk_extractor installer on Windows has two versions - one that installs just the command-line tools, and one that installs the GUI as well.

**Comment by jsouthTo** [comment]
Sorry for delay, tied up with some other things. Thanks for reminder on the platform independence of jars, I wasn’t sure if that would apply to arm as I only used x86 back then…last time I coded in java I had to use an actual paper book on servlets as reference, so it’s been a while...

I hope this isn’t something super obvious…I’ve used bulk_extractor plenty of times and never run into issues…pictures attached.

The system disk/data is both where I’m running bulk extractor and where the dmg file is located.
On May 16, 2022, 12:24 PM -0400, Simson L. Garfinkel ***@***.***>, wrote:
	Please let me know more details of the disk write error. Can you run df, then run bulk_extractor, then run df again, and post the results here (or email them to me), and also post the report.xml file (or email it to me). The xml file has in it the name of the disk image but it doesn't have any information extracted from the disk image. It's easy to validate by eye if you want to look at it.
Java jar files are both OS and CPU independent — that is the point of java. At least, they are always independent unless there is use of JNI (Java Native Interface). JAR files compiled in the past still run today, but old java code may be hard to get to compile.
The bulk_extractor installer on Windows has two versions - one that installs just the command-line tools, and one that installs the GUI as well.
